### PR TITLE
IBX-5273 - Order management API

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | IBX-5273 |
| Versions      | 4.5 |
| Edition       | Commerce |

Add a description of the API used to manage order information.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/commerce/order_management/order_management.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+description: The order management component covers creating orders, and managing their lifecycle.
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management.md:2]:
```suggestion
description: The order management component covers creating orders and managing their lifecycle.
```

File: docs/commerce/order_management/order_management.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Depending on their role, users can also track the status of their orders, review order details, cancel orders, issue invoices for orders, and generate aggregate reports.
+
+From the development perspective, the component enables customization of the order management workflow, integration with external systems to exchange order information.
+
+The component exposes the following:
+
+- [PHP API](order_management_api.md) that allows for managing orders
+- [REST API](../../api/rest_api/rest_api_reference/rest_api_reference.html#managing-orders) that helps get order information over HTTP
+
+/ ...
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management.md:18]: Something missing here

File: docs/commerce/order_management/order_management.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+description: The order management component covers creating orders, and managing their lifecycle.
+edition: commerce
+---
+
+# Order management
+
+The order management component enables users to search for orders, and filter search results.
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management.md:8]:
```suggestion
The order management component enables users to search for orders and filter search results. 
```

File: docs/commerce/order_management/order_management.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+description: The order management component covers creating orders, and managing their lifecycle.
+edition: commerce
+---
+
+# Order management
+
+The order management component enables users to search for orders, and filter search results. 
+Depending on their role, users can also track the status of their orders, review order details, cancel orders, issue invoices for orders, and generate aggregate reports.
+
+From the development perspective, the component enables customization of the order management workflow, integration with external systems to exchange order information.
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management.md:11]:
```suggestion
From the development perspective, the component enables customization of the order management workflow and integration with external systems to exchange order information.
```

File: docs/commerce/order_management/order_management.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+The component exposes the following:
+
+- [PHP API](order_management_api.md) that allows for managing orders
+- [REST API](../../api/rest_api/rest_api_reference/rest_api_reference.html#managing-orders) that helps get order information over HTTP
+
+/ ...
+
+### Order management service 
+
+The Cart package provides `Ibexa\Contracts\OrderManagement\OrderServiceInterface` Symfony service,
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management.md:22]:
```suggestion
The Cart package provides the `Ibexa\Contracts\OrderManagement\OrderServiceInterface` service, 
```

File: docs/commerce/order_management/order_management_api.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+### Get single order by identifier
+
+To access a single order by using its string identifier, use the `OrderService::getOrderByIdentifier` method:
+
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 50, 54) =]]
+```
+
+### Get single order by id
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management_api.md:24]:
```suggestion
### Get single order by ID
```

File: docs/commerce/order_management/order_management_api.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+To access a single order by using its numerical id, use the `OrderService::getOrder` method:
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management_api.md:26]:
```suggestion
To access a single order by using its numerical ID, use the `OrderService::getOrder` method:
```

File: docs/commerce/order_management/order_management_api.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 56, 61) =]]
+```
+
+## Get multiple orders
+
+To fetch multiple orders, use the `OrderService::findOrders` method. 
+It follows the same search Query pattern as other APIs:
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management_api.md:35]:
```suggestion
It follows the same search query pattern as other APIs:
```

File: docs/commerce/order_management/order_management_api.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 10, 15) =]]
+
+// ...
+[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 79, 88) =]]
+```
+
+## Create order
+
+To create an order, use the `orderServiceInterface::createOrder` method and provide
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management_api.md:46]:
```suggestion
To create an order, use the `OrderService::createOrder` method and provide 
```

File: docs/commerce/order_management/order_management_api.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+To create an order, use the `orderServiceInterface::createOrder` method and provide 
+it with the `Ibexa\Contracts\OrderManagement\Value\OrderCreateStruct` that contains a list of products, purchased quantities, product and total prices, as well as tax amounts.
+
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/OrderCommand.php', 63, 70) =]]
+```
+
+## Update order
+
+You can update order after the order is created.
```
Comment (DominikaK) [inline docs/commerce/order_management/order_management_api.md:55]:
```suggestion
You can update an order after it is created. 
```

**Review by DominikaK — CHANGES_REQUESTED** [review]
Additionally please run cs-fixer and verify that the command works as a whole

File: code_samples/api/commerce/src/Command/OrderCommand.php [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+use Ibexa\Contracts\OrderManagement\OrderServiceInterface;
+use Ibexa\Contracts\OrderManagement\Value\OrderQuery;
+use Ibexa\Contracts\OrderManagement\Value\Order\Query\Criterion\CompanyNameCriterion;
+use Ibexa\Contracts\OrderManagement\Value\Order\Query\Criterion\CustomerNameCriterion;
+use Ibexa\Contracts\OrderManagement\Value\Order\Query\Criterion\IdentifierCriterion;
+use Ibexa\Contracts\ProductCatalog\Values\Common\Query\Criterion\LogicalOr;
+use Ibexa\Contracts\OrderManagement\Value\OrderCreateStruct;
+use Ibexa\Contracts\OrderManagement\Value\OrderUpdateStruct;
+use Ibexa\Contracts\Security\Permission\PermissionResolver;
```
Comment (konradoboza) [inline code_samples/api/commerce/src/Command/OrderCommand.php:18]:
```suggestion
use Ibexa\Contracts\Core\Repository\PermissionResolver;
```

File: code_samples/api/commerce/src/Command/OrderCommand.php [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+use Ibexa\Contracts\Security\User\UserService;
```
Comment (konradoboza) [inline code_samples/api/commerce/src/Command/OrderCommand.php:19]:
```suggestion
use Ibexa\Contracts\Core\Repository\UserService;
```

File: docs/commerce/order_management/order_management.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
+---
+description: The order management component covers creating orders and managing their lifecycle.
+edition: commerce
+---
+
+# Order management
+
+The order management component enables users to search for orders,and filter search results.
```
Comment (konradoboza) [inline docs/commerce/order_management/order_management.md:8]:
```suggestion
The order management component enables users to search for orders and filter search results. 
```

File: docs/commerce/order_management/order_management.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+From the development perspective, the component enables customization of the order management workflow and integration with external systems to exchange order information.
+
+The component exposes the following:
+
+- [PHP API](order_management_api.md) that allows for managing orders
+- [REST API](../../api/rest_api/rest_api_reference/rest_api_reference.html#managing-orders) that helps get order information over HTTP
+
+### Order management service 
+
+The Cart package provides the `Ibexa\Contracts\OrderManagement\OrderServiceInterface` service,
```
Comment (konradoboza) [inline docs/commerce/order_management/order_management.md:20]: Wrong case of copy&paste.
```suggestion
The Order Management package provides the `Ibexa\Contracts\OrderManagement\OrderServiceInterface` service, 
```

**Review by konradoboza — CHANGES_REQUESTED** [review]

**Review by konradoboza — APPROVED** [review]

**Pull request merged**